crypto/src/main/java/org/cryptimeleon/incentive/crypto/model/SpendRequest.java

```diff
@@ -75,7 +76,8 @@ public SpendRequest(Representation repr, IncentivePublicParameters pp, FiatShami
         this.cTrace0 = groupG1.restoreVector(listRepr.get(6));
         this.cTrace1 = groupG1.restoreVector(listRepr.get(7));
 
-        var gamma = Util.hashGamma(zn, dsid, tid, cPre0, cPre1);
```

Comment:
This method needs the userChoice input, correct?

Reply:
Yes, thanks for pointing out.